lib/msal-core/test/UserAgentApplication.spec.ts

```diff
@@ -1281,6 +1292,17 @@ describe("UserAgentApplication.ts Class", function () {
             expect(acquireTokenPromise instanceof Promise).to.be.true;
             acquireTokenPromise.catch(error => {});
         });
+
+        it("throws an error if configured with a null request", () => {
+            let correctError;
+            try {
+                msal.acquireTokenPopup();
```

Comment:
I think you need to await this?

Reply:
Or you should, to be safe.

Reply:
hmm the error will happen before the promise right?

Reply:
so there is no return type

Reply:
in the scenario it doesnt throw before the promise train goes off on its own, I think that it is good that it continues on to the non catch expect that will fail

Reply:
That's not the point. Since its a promise, it will happen async, and in theory the error could happen and not get caught by the catch (I'm surprised what you have works). Read in the yellow in [this article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Rewriting_a_promise_chain_with_an_async_function) on the differences between `return await promise()` and `return promise()` when used in a try/catch.

Reply:
I disagree, as that would be potentially a false negative. Adding await ensures your catch will be executed if the promise throws.

e.g.
```js
try {
  // we dont care about the return value of the resolve promise
  await msal.acquireTokenPopup();
} catch (e) {
 // this will always get run if the promise throws
}
```

Reply:
Ah so I misread that you put the throws before/outside of the `new Promise`. I would personally like to see the assertion inside the promise.